lib/github/middleware.js

```diff
@@ -46,17 +45,22 @@ module.exports = function middleware (callback) {
       repo: (context.payload.repository) ? context.repo() : undefined
     })
 
+    const gitHubInstallationId = context.payload.installation.id
+    context.log = context.log.child({ gitHubInstallationId })
+
     if (context.payload.sender.type === 'Bot') {
+      context.log({ noop: 'bot', botId: context.payload.sender.id, botLogin: context.payload.sender.login }, 'Halting futher execution since the sender is a bot')
```

Comment:
@tebriel 

I figured this out with @JasonEtco's help. I added this blurb into the codebase to simulate the "bot doing stuff":

```js
  robot.on('issues.opened', async context => context.github.issues.createComment({
    ...context.issue(),
    title: 'Kaboom',
    body: 'Kaboom body'
  }))
```

More specifically this uses https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-an-installation and https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app but I didn't bother testing this out with cURL with that blurb.

Here's an example snippet from the payload:

```json
"sender": {
  "login": "jira-zmoazeni[bot]",
  "id": 55260329,
  "node_id": "MDM6Qm90NTUyNjAzMjk=",
  "avatar_url": "https://avatars0.githubusercontent.com/u/314?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D",
  "html_url": "https://github.com/apps/jira-zmoazeni",
  "followers_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/followers",
  "following_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/following{/other_user}",
  "gists_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/gists{/gist_id}",
  "starred_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/starred{/owner}{/repo}",
  "subscriptions_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/subscriptions",
  "organizations_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/orgs",
  "repos_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/repos",
  "events_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/events{/privacy}",
  "received_events_url": "https://api.github.com/users/jira-zmoazeni%5Bbot%5D/received_events",
  "type": "Bot",
  "site_admin": false
}
```

And example log message we will log:

```
INFO event: Halting futher execution since the sender is a bot (id=fb08cd80-6ae3-11ea-9f69-da238303cda2, gitHubInstallationId=7436896, noop=bot, botId=55260329, botLogin=jira-zmoazeni[bot])
```